m'mrm'mlrr DNA project

aleksey-ios-dev opened the pull request.

achernish85 committed:
DNA project

achernish85 committed:
project

achernish85 committed:
999

achernish85 committed:
нн

achernish85 committed:
m'mrm'mlrr DNA project

Sachs commented:
- Мутацию элемента ДНК можно сделать без лишнего перебора в while
- Категорию лучше вынести в отдельные файлы (New->File->"Objective-C category", там же выбрать от какого класса категория и как её назвать)

achernish85 committed:
Изменения и дополнения

Теперь нуклеотид для мутации не выбирается случайно "до победы", а
выбирается "наверняка" из усеченного массива.

Также вынес категорию в отдельный файл. Красиво!

achernish85 committed:
FINAL

Sachs commented:
Каждый раз в цикле создавать усечённый массив... ну вообще так себе решение, если представить, что массив в реальной задаче может быть не из 4х элементов, да и цепочка ДНК длиной явно не 100, тогда эта лишняя операция с памятью съест очень много времени в сумме

Sachs commented:
Вместо random(), посмотри описание функций arc4random(void) и arc4random_uniform(u_int32_t upper_bound), последняя выбирает случайное из диапазона [0, upper_bound - 1]; они не требуют рандомайза

Sachs commented:
(Не по теме кода) у тебя ссылка в "Assignment" ведёт на коммит, а не на пулл реквест, поправь
